File: cgi-bin/get.py
#!/usr/bin/python)

# Import modules for CGI handling
import cgi

# Create instance of FieldStorage
form = cgi.FieldStorage()
print(form)

# # Get data from fields
# user_name = form.getvalue('user_name')
# user_message  = form.getvalue('user_message')

# print ("<html>")
# print ("<head>")
# print ("<title>Webserv CGI</title>")
# print ("</head>")
# print ("<body>")
# print ("<h2>Hello %s</h2>" % (user_name,))
# print ("<p>Le message est : %s</p>" % (user_message))
# print ("</body>")
# print ("</html>")

File: cgi-bin/get.pl
local ($buffer, @pairs, $pair, $name, $value, %FORM);
# Read in text
$ENV{'REQUEST_METHOD'} =~ tr/a-z/A-Z/;

if ($ENV{'REQUEST_METHOD'} eq "GET") {
   $buffer = $ENV{'QUERY_STRING'};
}

# Split information into name/value pairs
@pairs = split(/&/, $buffer);

foreach $pair (@pairs) {
   ($name, $value) = split(/=/, $pair);
   $value =~ tr/+/ /;
   $value =~ s/%(..)/pack("C", hex($1))/eg;
   $FORM{$name} = $value;
}

$user_name = $FORM{user_name};
$user_message  = $FORM{user_message};

print "<html>";
print "<head>";
print "<title>Hello.pr</title>";
print "</head>";
print "<body>";
print "<h2>Hello $user_name </h2>";
print "<p>Le message est : $user_message </p>";
print "</body>";
print "</html>";

1;
